src/main/java/com/aws/greengrass/easysetup/DeviceProvisioningHelper.java

```diff
@@ -261,15 +269,45 @@ public void cleanThing(IotClient client, ThingInfo thing, boolean deletePolicies
     }
 
     /*
-     * Download root CA to a local file if the file does not exist
+     * Download root CA to a local file.
+     *
+     * To support HTTPS proxies and other custom truststore configurations, append to the file if it exists.
      */
-    private void downloadRootCAToFile(File f) throws IOException {
+    private void downloadRootCAToFile(File f) {
         if (f.exists()) {
-            outStream.printf("Root CA found at \"%s\". Skipping download.%n", f);
-            return;
+            outStream.printf("Root CA file found at \"%s\". Contents will be preserved.%n", f);
         }
         outStream.printf("Downloading Root CA from \"%s\"%n", ROOT_CA_URL);
-        downloadFileFromURL(ROOT_CA_URL, f);
+        try {
+            downloadFileFromURL(ROOT_CA_URL, f);
+            removeDuplicateCertificates(f);
+        } catch (IOException e) {
+            // Do not block as the root CA file may have been manually provisioned
+            outStream.printf("Failed to download Root CA - %s%n", e);
+        }
+    }
+
+    private void removeDuplicateCertificates(File f) {
```

Comment:
Is this just an optimization? If so, should we just let the original pem file be if it fails for whatever reason? i.e. line 308?

Reply:
We used to skip the CA download step if the root CA file existed, but if the root CA file contains the proxy CA certificate, it won't download the amazon root CA. So we want to always attempt the download and if it succeeds, remove duplicates (in case provisioning is run multiple times), and if it fails, log the error and not block the rest of provisioning as the device may have been manually setup.